### PR TITLE
test: add testcases to cover ClusterQueueImpl methods

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Review by ahg-g — COMMENTED** [review]
I am in favor of doing something similar to https://github.com/kubernetes-sigs/kueue/blob/a92130e5113090bad8add8a9df8b6d3f0eeb2ceb/pkg/cache/cache_test.go#L100

one tabulated test with operations and expectations, unless it is not easy to do.

**Comment by kerthcet** [comment]
> I am in favor of doing something similar to
> 
> https://github.com/kubernetes-sigs/kueue/blob/a92130e5113090bad8add8a9df8b6d3f0eeb2ceb/pkg/cache/cache_test.go#L100
> 
> one tabulated test with operations and expectations, unless it is not easy to do.

That's what I thought at first, then I realized it's procedure-oriented not scenario-oriented just like heap_test.go, so I changed it this way.

File: pkg/queue/cluster_queue_impl_test.go [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+	newWorkload := func(name string, t metav1.Time) *kueue.Workload {
+		wl := &kueue.Workload{}
+		wl.Name = name
+		wl.CreationTimestamp = t
+		wl.Namespace = defaultNamespace
+		return wl
+	}
+
+	wl1 := newWorkload("workload-1", metav1.NewTime(now))
+	wl2 := newWorkload("workload-2", metav1.NewTime(now.Add(time.Second)))
```
[the remark this answers is not in the document]
Comment (kerthcet) [reply pkg/queue/cluster_queue_impl_test.go:73]: It has already been implemented as `Creation`

File: pkg/queue/cluster_queue_impl_test.go [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+	newWorkload := func(name string) *kueue.Workload {
+		wl := &kueue.Workload{}
+		wl.Name = name
+		wl.Namespace = defaultNamespace
+		return wl
+	}
+
+	// Add workloads to cluster queue.
+	wl1 := newWorkload("workload-1")
+	wl2 := newWorkload("workload-2")
```
[the remark this answers is not in the document]
Comment (kerthcet) [reply pkg/queue/cluster_queue_impl_test.go:108]: got it.

File: pkg/queue/cluster_queue_impl_test.go [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+		t.Error("cluster queue should have two workload")
+	}
+
+	cq.Delete(wl1)
+	if cq.Pending() != 1 {
+		t.Error("cluster queue should have only one workload")
+	}
+
+	// Change workload item, ClusterQueue.Delete should only care about the namespace and name.
+	wl2.Spec = kueue.WorkloadSpec{QueueName: "default"}
```
[the remark this answers is not in the document]
Comment (kerthcet) [reply pkg/queue/cluster_queue_impl_test.go:122]: We used to call `delete` with `workload` obj as parameter passed through to `heap.Delete`, then I changed to delete with workload key, so I added this test.

**Comment by kerthcet** [comment]
addressed all the comments @ahg-g

**Review by ahg-g — COMMENTED** [review]
thanks, pls squash

**Comment by kerthcet** [comment]
squashed.

**Pull request merged**